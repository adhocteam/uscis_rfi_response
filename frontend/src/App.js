import React, { Component } from "react";
import "./App.css";

class App extends Component {
  constructor(props) {
    super(props);
    this.state = {
      reader: null
    };
    this.readFile = this.readFile.bind(this);
  }

  // stuck trying to set the state with the bytes, so i just put a fetch
  // call to send it to the server.
  readFile() {
    var preview = document.querySelector("img");

    var file = document.querySelector("input[type=file]").files[0];
    var reader = new FileReader();

    reader.addEventListener(
      "load",
      function() {
        preview.src = reader.result;
        console.log(reader.result);

        fetch("/api/upload", {
          method: "post",
          headers: {
            Accept: "application/json, text/plain, */*",
            "Content-Type": "application/json"
          },
          body: JSON.stringify({
            image_name: file.name,
            image_base64: reader.result,
            timestamp: new Date().toISOString()
          })
        });
      },
      false
    );

    if (file) {
      reader.readAsDataURL(file);
    }
  }
  render() {
    return (
      <div className="App">
        <input className="Input" type="file" onChange={this.readFile} />
        <h3> image preview: </h3>
        <img src="" height="200" alt="Image preview..." />
      </div>
    );
  }
}

export default App;
