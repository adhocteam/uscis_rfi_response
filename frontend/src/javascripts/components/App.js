import React from "react";
import { BrowserRouter, Route, Switch } from "react-router-dom";
import AdminPage from "./AdminPage";
import UploadPage from "./UploadPage";

export default function App() {
  return (
    <BrowserRouter>
      <div>
        <div className="ds-u-fill--primary-darkest">
          <div className="ds-l-container ds-u-md-padding-top--1">
            <div className="ds-l-row">
              <div className="ds-l-col">
                <h1
                  id="main"
                  className="ds-u-serif ds-u-color--white ds-u-font-size--h2 ds-u-md-font-size--h1 ds-u-font-weight--normal ds-u-margin-top--3 ds-u-margin-bottom--3"
                >
                  USCIS Uploader Front-End
                </h1>
              </div>
            </div>
          </div>
        </div>
        <Switch>
          <Route path="/" exact component={UploadPage} />
          <Route path="/review" component={AdminPage} />
        </Switch>
      </div>
    </BrowserRouter>
  );
}
