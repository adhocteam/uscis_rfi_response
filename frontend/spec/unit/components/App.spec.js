import React from "react";
import { shallow } from "enzyme";
import App from "components/App.js";

test("renders", () => {
  const wrapper = shallow(<App />);
  expect(wrapper).toMatchSnapshot();
});
