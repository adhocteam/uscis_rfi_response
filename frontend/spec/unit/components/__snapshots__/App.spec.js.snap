// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders 1`] = `
<Router
  history={
    Object {
      "action": "POP",
      "block": [Function],
      "createHref": [Function],
      "go": [Function],
      "goBack": [Function],
      "goForward": [Function],
      "length": 1,
      "listen": [Function],
      "location": Object {
        "hash": "",
        "pathname": "blank",
        "search": "",
        "state": undefined,
      },
      "push": [Function],
      "replace": [Function],
    }
  }
>
  <div>
    <div
      className="ds-u-fill--primary-darkest"
    >
      <div
        className="ds-l-container ds-u-md-padding-top--1"
      >
        <div
          className="ds-l-row"
        >
          <div
            className="ds-l-col"
          >
            <h1
              className="ds-u-serif ds-u-color--white ds-u-font-size--h2 ds-u-md-font-size--h1 ds-u-font-weight--normal ds-u-margin-top--3 ds-u-margin-bottom--3"
              id="main"
            >
              USCIS RFDS Front-End
            </h1>
          </div>
        </div>
      </div>
    </div>
    <Switch>
      <Route
        component={[Function]}
        exact={true}
        path="/login"
      />
      <Route
        component={[Function]}
        exact={true}
        path="/logout"
      />
      <Route
        component={[Function]}
        path="/review"
      />
      <Route
        component={[Function]}
        path="/request"
      />
      <Route
        component={[Function]}
        exact={true}
        path="/"
      />
    </Switch>
  </div>
</Router>
`;
